### PR TITLE
update oxog tool and add metadata-parser tool, add test cases

### PR DESCRIPTION
**Pull request by lindaxiang**

### CODE REVIEW

File: dna-seq-alignment/main.nf [hunk 1]
```diff
@@ -301,11 +303,15 @@ workflow DnaAln {
         splitItvls(gatkCollectOxogMetrics_params.oxog_scatter, file(ref_genome_fa),
             Channel.fromPath(getSIIdx(ref_genome_fa), checkIfExists: true).collect(), file('NO_FILE'))
 
+        // get single/paired end info
+        mParser(analysis_metadata)
+     
+        println(mParser.out.paired)
```
Comment (junjun-zhang) [inline dna-seq-alignment/main.nf:309]: remove this?

Comment (lindaxiang) [reply dna-seq-alignment/main.nf:309]: oh, right. good catch.

File: dna-seq-alignment/tests/test-local_2bams.1.json [hunk 2]
```diff
@@ -14,5 +14,6 @@
   },
   "cpus": 2,
   "mem": 2,
-  "tempdir": "NO_DIR"
+  "tempdir": "NO_DIR",
+  "publish_dir": "publish_dir"
```
Comment (junjun-zhang) [inline dna-seq-alignment/tests/test-local_2bams.1.json:18]: something minor, but can you change the value of publish_dir to `outdir`?  `outdir` is already included in `.gitignore` file, so it won't be accidentally commited to git.

Comment (lindaxiang) [reply dna-seq-alignment/tests/test-local_2bams.1.json:18]: Sure. Thank you for letting me know. I will make the change.

**Comment by junjun-zhang** [comment]
would you be able to test this in the qa env?

**Comment by lindaxiang** [comment]
> would you be able to test this in the qa env?

I have tested it and completed successfully in `rdpc-qa`. @junjun-zhang

**Review by junjun-zhang — APPROVED** [review]
looking good and test passed in QA

**Pull request merged**